### PR TITLE
Gnomad keys

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality or fixes a bug.

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [x] tests executed by DN

### CODE REVIEW

**Comment by dnil** [comment]
I'll leave this like this until we have a good example - I just realised the `poorpipe` genomes were annotated with 1000g only.

**Comment by dnil** [comment]
I think we got a sample with annotations now - it’s been delayed a bit due to higher than usual error rates. Will check!

**Comment by dnil** [comment]
Finally some files passed to annotation again! While perhaps not the finished product, good enough to test this PR:
Screenshot 2024-04-29 at 18 18 44

**Comment by northwestwitch** [comment]
@dnil just assign me if you think this is ready for a review

**Comment by dnil** [comment]
> @dnil just assign me if you think this is ready for a review

Anytime it's not draft, though things may still change with second or third thought. 😉 I made a small refactor to the parser which was just over the CodeFactor complexity threshold, though not the SonarCloud one yet. I also retested it again on `winningchicken`, and it seemed to work ok still.

File: scout/parse/variant/frequency.py [hunk 1]
```diff
@@ -27,8 +27,8 @@ def parse_frequencies(variant, transcripts):
     exac_max_keys = ["ExAC_MAX_AF", "EXAC_MAX_AF"]
 
     # Gnomad have both snv and sv frequencies
-    gnomad_keys = ["GNOMADAF", "GNOMAD_AF", "gnomad_svAF"]
-    gnomad_max_keys = ["GNOMADAF_popmax", "GNOMADAF_POPMAX", "GNOMADAF_MAX"]
+    gnomad_keys = ["GNOMADAF", "GNOMAD_AF", "gnomADg_AF", "gnomad_svAF"]
```
Comment (northwestwitch) [inline scout/parse/variant/frequency.py:30]: Perhaps make them constants, like GNOMAD_KEYS. No need to move them under contants I think, they're more readable here.

File: scout/parse/variant/transcript.py [hunk 2]
```diff
@@ -312,30 +316,32 @@ def set_variant_frequencies(transcript, entry):
...
 
             value = entry[key]
-            if not value:
+            if not value or value == ".":
                 continue
 
-            # This is the 1000G max af information
-            if key == "AF" or key == "1000GAF":
+            # This is the 1000G max AF information
+            if key in ["AF", "1000GAF", "1000GP3_AF"]:
```
Comment (northwestwitch) [inline scout/parse/variant/transcript.py:327]: Add them into a constant instead?

File: scout/parse/variant/transcript.py [hunk 3]
```diff
@@ -312,30 +316,32 @@ def set_variant_frequencies(transcript, entry):
...
                 transcript["thousand_g_maf"] = float(value)
                 continue
 
-            if key == "GNOMAD_AF":
-                transcript["gnomad_maf"] = float(value)
-                continue
+            # gnomAD. Use plain (older) AF if available. For a secondary choice, prefer genomes over exomes.
+            for gnomad_ordered_key in ["GNOMAD_AF", "GNOMADG_AF", "GNOMAD_EXOMES_AF"]:
```
Comment (northwestwitch) [inline scout/parse/variant/transcript.py:332]: Same here, I would declare a constant and use that instead

**Review by northwestwitch — APPROVED** [review]
Looks good to me. Only a couple of suggestions.

**Pull request merged**